docs/website/docs/plus/getting-started/transformations.md

```diff
@@ -0,0 +1,446 @@
+def employees():
+    yield [
+        {"id": 1, "name": "Alice", "department_id": 1, "office_id": 1},
+        {"id": 2, "name": "Bob", "department_id": 2, "office_id": 2},
+        {"id": 3, "name": "Charlie", "department_id": 1, "office_id": 2},
+        {"id": 4, "name": "Diana", "department_id": 2, "office_id": 1},
+    ]
+
+# adds a custom hint to the name column, we will use this to test lineage later
+employees.apply_hints(columns={"name": {"x-pii": True}})  # type: ignore
+
+# Payroll data
+@dlt.resource(write_disposition="replace")
+def payroll():
+    yield [
+        {"id": 1, "employee_id": 1, "month": "2024-01", "amount": 7000},
+        {"id": 2, "employee_id": 1, "month": "2024-02", "amount": 7100},
+        {"id": 3, "employee_id": 2, "month": "2024-01", "amount": 5500},
+        {"id": 4, "employee_id": 2, "month": "2024-02", "amount": 5600},
+        {"id": 5, "employee_id": 3, "month": "2024-01", "amount": 5000},
+        {"id": 6, "employee_id": 3, "month": "2024-02", "amount": 5100},
+        {"id": 7, "employee_id": 4, "month": "2024-01", "amount": 6000},
+        {"id": 8, "employee_id": 4, "month": "2024-02", "amount": 6100},
+    ]
+
+# Create a DLT source with three resources
+@dlt.source
+def employee_data():
+    return offices(), employees(), payroll(), departments()
+
+# Create and run pipeline
+company_pipeline = dlt.pipeline(
+    pipeline_name="company_pipeline",
+    destination="duckdb",
+    dataset_name="company_dataset"
+)
+
+# Execute pipeline
+info = company_pipeline.run(employee_data())
+print(info)
+```
+</details>
+
+We have a straightforward company database with live employee master data containing names and payroll payments, as well as two tables for offices and departments. As always, you can inspect the contents of the dataset by running `dlt pipeline company_pipeline show` to open the dlt streamlit app.
+
+
+## A first few transformations 
+
+Transformations work very similarly to regular dlt.resources, which they are under the hood, and can be grouped into a regular `dlt.source`. Transformations take a populated dataset as input to build transformations from this source and move the computed data into the same or another dataset. If data stays within the same physical destination, you can execute the transformation as a SQL statement and no data needs to be moved to the machine where the dlt pipeline is running. For all other cases, data will be extracted the same way as with a regular dlt pipeline.
+
+Let's move some data from the company dataset into a fictional warehouse dataset. Let's assume we want the employee master data to be available in the warehouse dataset. We want to make sure not to include employee names in the warehouse dataset for privacy reasons.
+
+```python
+
+# get the company dataset
+company_dataset = company_pipeline.dataset()
+
+# print the employee table without the name column as arrow table
+print(company_dataset.employees.drop("name").arrow())
```

Comment:
I don't like the arrow table output. df() looks much nice imo
```suggestion
print(company_dataset.employees.drop("name").df())
```
![image](https://github.com/user-attachments/assets/61e2efad-70fe-4a75-af2e-04277e024a5e)